lib/components/topic_button.dart

```diff
@@ -1,26 +1,67 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter_svg/flutter_svg.dart';
+import 'package:maui/db/dao/quiz_progress_dao.dart';
+import 'package:maui/db/entity/quiz_progress.dart';
+import '../repos/quiz_progress_repo.dart';
 
-class TopicButton extends StatelessWidget {
+class TopicButton extends StatefulWidget {
   final String text;
   final int color;
   final String image;
   final VoidCallback onPress;
+  
+
   TopicButton({
     Key key,
     @required this.text,
     this.color,
     this.image,
-    this.onPress,
+    this.onPress
 //      this.sizeHieght,
 //      this.sizeWidth
   }) : super(key: key);
+  
+  @override
+  State createState() => new TopicButtonState();
+
+}
+
+class TopicButtonState extends State<TopicButton>{
+  QuizProgress _quizProgress;
+  bool _isLoading = true;
+  var doneStatus;
+
+  @override
+  void initState() {
+    super.initState();
+    _initData();
+  }
+
+  void _initData() async{
+    setState(() => _isLoading = true);
+   _quizProgress =
+        await QuizProgressRepo().getQuizProgressByTopicId("1");    
+    doneStatus = (double.parse(_quizProgress.done)/6);  
```

Comment:
Obtain the total number of quizzes for this topic and use that instead of hard coding